task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -346,11 +348,15 @@ def get_relevant_upstream_map_indexes(
 def _xcom_push(ti: RuntimeTaskInstance, key: str, value: Any, mapped_length: int | None = None) -> None:
     # Private function, as we don't want to expose the ability to manually set `mapped_length` to SDK
     # consumers
-    from airflow.models.xcom import XCom
+    from airflow.serialization.serde import serialize
 
     # TODO: Move XCom serialization & deserialization to Task SDK
     #   https://github.com/apache/airflow/issues/45231
-    value = XCom.serialize_value(value)
+
+    # The execution API server now deals in json compliant objects.
+    # It is responsibility of the client to handle any non native object serialization.
+    # serialize does just that.
+    value = serialize(value)
```

Comment:
fyi: `XCom.serialize_value` uses `airflow.serialization.serde.serialize` and does a little more that is also used in `XComObjectStorageBackend`

https://github.com/apache/airflow/blob/8f63b828ace09b1095229f22b0c6d1f0f85ea81a/airflow/utils/json.py#L87-L105

https://github.com/apache/airflow/blob/main/providers/common/io/src/airflow/providers/common/io/xcom/backend.py#L73

Reply:
This is what will impact Custom XCom backend, previously, `XCom.serialiaze_value` always returned string because it called `json.dumps`:

https://github.com/apache/airflow/blob/8f63b828ace09b1095229f22b0c6d1f0f85ea81a/airflow/models/xcom.py#L443-L458

We should clarify that in a newsfragment imo

Reply:
For the custom xcom backend, we anyways store ONLY the `serialised` path in the DB right?
Even with this change, we will store the same but not in a serialised manner, so those interactions will now happen via python strings. That should still be OK i guess

Reply:
Resolving for now to merge, we can discuss in a follow up